sys/include/tasklet_param.h

```diff
@@ -0,0 +1,35 @@
+
+#include "tasklet.h"
+
+#ifdef MODULE_TASKLET_HIGHPRIO
+static char _stack_t1[SOME_STACK_SIZE];
+#endif
+
+#ifdef MODULE_TASKLET_LOWPRIO
+static char _stack_t2[SOME_OTHER_STACK_SIZE];
+
+/* syntactic sugar: we could give some names - however, we need to find a better
+ * way to do this non statically depending on the actual selected modules... */
+#define TASKLET_LOWPRIO         (1u) /* position in params array */
```

Comment:
You could do something like this:
```
#define TASKLET_PRIO_LOWEST (0U)

#ifdef MODULE_TASKLET_PRIO_LOW
#define TASKLET_PRIO_LOW (TASKLET_PRIO_LOWEST + 1)
#else
#define TASKLET_PRIO_LOW (TASKLET_PRIO_LOWEST)
#endif

#ifdef MODULE_TASKLET_PRIO_MEDIUM
#define TASKLET_PRIO_MEDIUM (TASKLET_PRIO_LOW + 1)
#else
#define TASKLET_PRIO_LOW (TASKLET_PRIO_LOW)
#endif

#ifdef MODULE_TASKLET_PRIO_HIGH
#define TASKLET_PRIO_HIGH (TASKLET_PRIO_MEDIUM + 1)
#else
#define TASKLET_PRIO_HIGH (TASKLET_PRIO_MEDIUM)
#endif

#ifdef MODULE_TASKLET_PRIO_HIGHEST
#define TASKLET_PRIO_HIGHEST (TASKLET_PRIO_HIGH + 1)
#else
#define TASKLET_PRIO_HIGHEST (TASKLET_PRIO_HIGH)
#endif

#define TASKLET_PRIO_NUMOF (TASKLET_PRIO_HIGHEST + 1)
```

One could argue that it is a bit confusing that the defines for all 5 priority levels are always defined. But I would say that this is actually an advantage, as I could decided later on that I do not need the divide in lowest and low without having to change all uses of `TASKLET_PRIO_LOW` to `TASKLET_PRIO_LOWEST`.

Reply:
As stated, please lets first focus on the overall concept before jumping into implementation details...